internal/dataplane/parser/translate_ingress.go

```diff
@@ -187,72 +188,80 @@ func (p *Parser) ingressRulesFromIngressV1() ingressRules {
 		result.SecretNameToSNIs.addFromIngressV1TLS(ingressSpec.TLS, ingress.Namespace)
 
 		var objectSuccessfullyParsed bool
-		for i, rule := range ingressSpec.Rules {
-			if rule.HTTP == nil {
-				continue
+
+		if p.featureEnabledCombinedServiceRoutes {
+			for _, kongStateService := range translators.TranslateIngress(ingress) {
+				result.ServiceNameToServices[*kongStateService.Service.Name] = *kongStateService
 			}
-			for j, rulePath := range rule.HTTP.Paths {
-				if strings.Contains(rulePath.Path, "//") {
-					log.Errorf("rule skipped: invalid path: '%v'", rulePath.Path)
+			objectSuccessfullyParsed = true
+		} else {
+			for i, rule := range ingressSpec.Rules {
```

Comment:
To my feeling, this block is a little bit lengthy. Perhaps we could extract this to a separate `func`? This was we also decrease the indentation and make it (arguably) more readble.

Reply:
Despite the Github diff, this block was NOT changed by this PR, it was only indented for conditionality based on whether the newer mode is enabled. You're not wrong that this whole area of code is lengthy and in fact I want to rewrite ALL of parser really badly. However for reasons of time and to avoid disrupting the existing code I'm intentionally trying to maintain a light touch on the existing code for the purposes of this optional feature. I would like to avoid refactoring what was there for now and allow a future iteration to be responsible for that. Let me know your thoughts.

Reply:
Fair enough. Let's leave it as is.

OTOH I'm afraid that this "refactoring" having such a low prio ( I assume ) will never get done 🙃

Reply:
Let us try this: created an issue based on this comment https://github.com/Kong/kubernetes-ingress-controller/issues/2515